src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -399,6 +414,166 @@ void ZeroInferRequest::set_tensor(const ov::Output<const ov::Node>& port, const
+        OPENVINO_ASSERT(item->get_shape()[batch_idx] == 1,
+                        "set_input_tensors/set_tensors. Tensors shall represent one item in a batch, ",
+                        item->get_shape()[batch_idx],
+                        " provided");
+    });
+    auto tensors_size = static_cast<int>(tensors.size());
+    if (input.get_partial_shape().rank().is_static()) {
+        OPENVINO_ASSERT(batch_idx >= 0 && batch_idx < input.get_partial_shape().rank().get_length(),
+                        "set_input_tensors/set_tensors error. Layout ",
+                        layout.to_string(),
+                        " is incorrect for operation with shape ",
+                        input.get_partial_shape());
+        auto batch = input.get_partial_shape()[batch_idx];
+
+        OPENVINO_ASSERT(batch.is_dynamic() || batch.get_length() == tensors_size,
+                        "set_input_tensors/set_tensors error. Input shape ",
+                        input.get_partial_shape(),
+                        "batch ",
+                        batch,
+                        "doesn't match with total blobs count: ",
+                        tensors_size);
+    }
+
+    auto batched_shape = tensors[BATCH_AXIS]->get_shape();
+    auto element_type = tensors[BATCH_AXIS]->get_element_type();
+    batched_shape[batch_idx] = tensors_size;
+    for (const auto& item : tensors) {
+        OPENVINO_ASSERT(item, "Unintialized tensor is provided!");
+        auto item_shape = item->get_shape();
+        item_shape[batch_idx] = batched_shape[batch_idx];
+        OPENVINO_ASSERT(item_shape == batched_shape && item->get_element_type() == element_type &&
+                            "set_input_tensors/set_tensors error. Tensor with element type ",
+                        item->get_element_type(),
+                        " and shape ",
+                        item_shape,
+                        " is not compatible with batched tensor with element type ",
+                        element_type,
+                        " and shape ",
+                        batched_shape);
+        OPENVINO_ASSERT(item->is_continuous(), "Strides for batched tensors should be default.");
+    }
+}
+
+void ZeroInferRequest::set_tensors(const ov::Output<const ov::Node>& port,
+                                   const std::vector<ov::SoPtr<ov::ITensor>>& tensors) {
+    OV_ITT_SCOPED_TASK(itt::domains::LevelZeroBackend, "set_tensors");
+    if (tensors.size() == 1) {
+        set_tensor(port, tensors[0]);
+        return;
+    }
+
+    auto foundPort = find_port(port);
+    OPENVINO_ASSERT(foundPort.found(), "Cannot find input tensor for port ", port);
+    if (!foundPort.is_input()) {
+        OPENVINO_THROW("set_input_tensors/set_tensors is not supported for output port.");
+    }
+
+    check_batched_tensors(port, tensors);
+
+    _userBatchedTensors[foundPort.idx] = tensors;
+
+    if (_initStructs->getMutableCommandListVersion()) {
+        if (_batchSize.has_value()) {
```

Comment:
Isn't it an error situation when `_batchSize` does not have value? Should it be signaled if so?

Reply:
Same question for `remoteTensor`. What if we receive `nullptr` after `dynamic_pointer_cast`

Reply:
And for branches
```
case ZE_MEMORY_TYPE_UNKNOWN:
case ZE_MEMORY_TYPE_FORCE_UINT32:
```

Reply:
nop isn't an error. Actually, this logic is only for the case when the bathing is handled by the plugin. When batching is handled by the compiler we need to do memcpy since we can not set different memory regions for the same input.

We can do this only in case of the plugin batching, since we are using multiple command lists for each bach.

Reply:
we are checking if the tensor is a remote or not. Also added func tests with this scenario.

Reply:
In case of those branches just don't do anything because the memory isn't part of the same l0 context